.github/workflows/ruff_format.yaml

```diff
@@ -0,0 +1,11 @@
+name: ruff format
```

Comment:
I don't think we want to format the code with ruff in pull request. Linting should happen and notify developer if errors are there. Developers should fix those themselves.

Reply:
This is not formatting code in the PR itself, just checking if it is formatted properly.

I've added a git hook which should be installed to verify this as well, would be good to keep the code formatted uniformly in the future.

Reply:
if a developer disagrees with the proposed formatting he can add format suppression as described here:

https://docs.astral.sh/ruff/formatter/#format-suppression